packages/core/src/common/command.ts

```diff
@@ -265,8 +265,6 @@ export class CommandRegistry implements CommandService {
 
     /**
      * Execute the active handler for the given command and arguments.
-     *
-     * Reject if a command cannot be executed.
```

Comment:
It changes semantics for all clients. We should try to reduce impact of changes, otherwise it is hard to accept PRs.

What is the reasoning behind it? Can we fix it in the plugin system in the place where we call `executCommand`?

Reply:
> What is the reasoning behind it?

The [extension](https://marketplace.visualstudio.com/items?itemName=vscodevim.vim) fails an error is thrown while executing a command.
> Can we fix it in the plugin system in the place where we call `executCommand`?

The `$executeCommand` method in the plugin system returns `PromiseLike` which doesn't allow to use `async / await`, so I can't exexute and catch the error there:
https://github.com/eclipse-theia/theia/blob/d5c81105393a811de3097ff06bdf64ab98ef3d80/packages/plugin-ext/src/main/browser/command-registry-main.ts#L75-L77

Reply:
Actually, I don't think we should change it at all. VS Code throws exceptions as well:
- https://github.com/microsoft/vscode/blob/d2c5d5e1184ca78d051813b17255568158bf8f3a/src/vs/workbench/services/commands/common/commandService.ts#L80
- https://github.com/microsoft/vscode/blob/d2c5d5e1184ca78d051813b17255568158bf8f3a/src/vs/workbench/api/browser/mainThreadCommands.ts#L81

If a command is missing we should fix it instead.

> The $executeCommand method in the plugin system returns PromiseLike which doesn't allow to use async / await, so I can't exexute and catch the error there:

You can change a return type to `Promise`. But it is relevant, logging is not correct behaviour.

Reply:
I found a way how to catch an error in the plugin system side. @akosyakov could you please check?

Reply:
Why do you want to do it? VS Code does not do it and we should not. VS Code extensions should expect an error if a command is missing and we should fix missing commands.

Reply:
@vinokurig hi, I don't understand as well why we should only log the error.

What is the command that is being executed, that is failing and that you're logging ?

Reply:
@akosyakov 
> VS Code extensions should expect an error if a command is missing and we should fix missing commands.

Yes, but VsCode doesn't throw an error if the command's handler is not enabled:
https://github.com/microsoft/vscode/blob/d386b49dee6267238389c0a616af2be89726a4e3/src/vs/editor/browser/editorExtensions.ts#L177-L180
@benoitf 
> What is the command that is being executed, that is failing and that you're logging ?

The command is `editor.action.wordHighlight.trigger` it's handler is not always active, so Theia throws an error when it's not active:
https://github.com/eclipse-theia/theia/blob/27745bf2483e10600978d9e7e4821a1a0ab3ac83/packages/core/src/common/command.ts#L272-L284

Reply:
Ok, makes sense for `editor.action.wordHighlight.trigger`. I don't think we can remove `isEnabled` predicate from the monaco commands contributed based on editor actions. It will break the quick command palette.

Let's go with your proposal. Please instead of doing string matching, add `code`  property to a created error object and match based on it, i.e.
- in commands:ts `throw Object.assign(new Error('...'), { code: 'NO_ACTIVE_HANDLER' });`
- in command-registry-main.ts `if ('code' in e && e['code'] === 'NO_ACTIVE_HANDLER')`

Reply:
Thanks, done